README.md

```diff
@@ -174,6 +176,42 @@ To be able to create your own data source implementation you need to implement t
 
 You can see a really simple data source implementation on our `mem` package.
 
+## Indexes
+
+`go-mysql-server` exposes a series of interfaces to allow you to implement your own indexes so you can speedup your queries.
+
+Taking a look at the main index interface:
+
+```go
+// Index is the basic representation of an index. It can be extended with
+// more functionality by implementing more specific interfaces.
+type Index interface {
+	// Get returns an IndexLookup for the given key in the index.
+	Get(key ...interface{}) (IndexLookup, error)
+	// Has checks if the given key is present in the index.
+	Has(key ...interface{}) (bool, error)
+	// ID returns the identifier of the index.
+	ID() string
+	// Database returns the database name this index belongs to.
+	Database() string
+	// Table returns the table name this index belongs to.
+	Table() string
+	// Expressions returns the indexed expressions. If the result is more than
+	// one expression, it means the index has multiple columns indexed. If it's
+	// just one, it means it may be an expression or a column.
+	ExpressionHashes() []ExpressionHash
+	// Driver ID of the index.
+	Driver() string
+}
+```
+
+You must note a couple of constraints:
+
+- this abstraction will index one or more columns.
+- if you want to index an expression you will only be able to index **one and only one** expression at a time.
```

Comment:
...if you want to index an expression **that is not a column**...

Reply:
maybe:
> This abstraction lets you create an index for multiple columns (one or more) or for **only one** expression (e.g. function applied on multiple columns).